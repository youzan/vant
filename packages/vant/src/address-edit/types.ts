import type { ComponentPublicInstance } from 'vue';
import type { AddressEditProps } from './AddressEdit';

export type AddressEditSearchItem = {
  name?: string;
  address?: string;
};

export type AddressEditInfo = {
  tel: string;
  name: string;
  city: string;
  county: string;
  province: string;
  areaCode: string;
  isDefault?: boolean;
  addressDetail: string;
};

export type AddressEditExpose = {
  setAreaCode: (code?: string | undefined) => void;
  setAddressDetail: (value: string) => void;
};

export type AddressEditInstance = ComponentPublicInstance<
  AddressEditProps,
  AddressEditExpose
>;

export type AddressEditThemeVars = {
  addressEditPadding?: string;
  addressEditButtonsPadding?: string;
  addressEditButtonMarginBottom?: string;
  addressEditButtonFontSize?: string;
};
