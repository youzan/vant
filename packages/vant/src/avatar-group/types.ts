export type AvatarGroupCascadingValue = 'left-up' | 'right-up';

export type AvatarGroupThemeVars = {
  avatarGroupOverlap?: string;
};
