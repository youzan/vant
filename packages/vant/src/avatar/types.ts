// Types
export type AvatarSize = 'large' | 'medium' | 'normal' | 'small';
export type AvatarShape = 'square' | 'round';

export type AvatarThemeVars = {
  avatarSize?: string;
  avatarSizeLarge?: string;
  avatarSizeMedium?: string;
  avatarSizeSmall?: string;
  avatarBgColor?: string;
  avatarTextColor?: string;
  avatarFontSize?: string;
  avatarFontSizeLarge?: string;
  avatarFontSizeMedium?: string;
  avatarFontSizeSmall?: string;
  avatarFontWeight?: string;
  avatarLineHeight?: string;
  avatarBorderRadius?: string;
};
