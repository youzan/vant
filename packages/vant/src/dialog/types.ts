import type { CSSProperties, TeleportProps } from 'vue';
import type { Interceptor, Numeric } from '../utils';

export type DialogTheme = 'default' | 'round-button';
export type DialogAction = 'confirm' | 'cancel';
export type DialogMessage = string | (() => JSX.Element);
export type DialogMessageAlign = 'left' | 'center' | 'right' | 'justify';

export type DialogOptions = {
  title?: string;
  width?: Numeric;
  theme?: DialogTheme;
  message?: DialogMessage;
  overlay?: boolean;
  teleport?: TeleportProps['to'];
  className?: unknown;
  allowHtml?: boolean;
  lockScroll?: boolean;
  transition?: string;
  beforeClose?: Interceptor;
  messageAlign?: DialogMessageAlign;
  overlayClass?: string;
  overlayStyle?: CSSProperties;
  closeOnPopstate?: boolean;
  showCancelButton?: boolean;
  showConfirmButton?: boolean;
  cancelButtonText?: string;
  cancelButtonColor?: string;
  cancelButtonDisabled?: boolean;
  confirmButtonText?: string;
  confirmButtonColor?: string;
  confirmButtonDisabled?: boolean;
  closeOnClickOverlay?: boolean;
  destroyOnClose?: boolean;
  keyboardEnabled?: boolean;
};

export type DialogThemeVars = {
  dialogWidth?: string;
  dialogSmallScreenWidth?: string;
  dialogFontSize?: string;
  dialogTransition?: string;
  dialogRadius?: string;
  dialogBackground?: string;
  dialogHeaderFontWeight?: string;
  dialogHeaderLineHeight?: number | string;
  dialogHeaderPaddingTop?: string;
  dialogHeaderIsolatedPadding?: string;
  dialogMessagePadding?: string;
  dialogMessageFontSize?: string;
  dialogMessageLineHeight?: number | string;
  dialogMessageMaxHeight?: string;
  dialogHasTitleMessageTextColor?: string;
  dialogHasTitleMessagePaddingTop?: string;
  dialogButtonHeight?: string;
  dialogRoundButtonHeight?: string;
  dialogConfirmButtonTextColor?: string;
};
