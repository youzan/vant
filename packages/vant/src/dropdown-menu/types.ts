import type { ComponentPublicInstance, Ref } from 'vue';
import type { DropdownMenuProps } from './DropdownMenu';

export type DropdownMenuDirection = 'up' | 'down';

export type DropdownMenuProvide = {
  id: string;
  props: DropdownMenuProps;
  offset: Ref<number>;
  opened: Ref<boolean>;
  updateOffset: () => void;
};

export type DropdownMenuExpose = {
  opened: Ref<boolean>;
  close: () => void;
};

export type DropdownMenuInstance = ComponentPublicInstance<
  DropdownMenuProps,
  DropdownMenuExpose
>;

export type DropdownMenuThemeVars = {
  dropdownMenuHeight?: string;
  dropdownMenuBackground?: string;
  dropdownMenuShadow?: string;
  dropdownMenuTitleFontSize?: string;
  dropdownMenuTitleTextColor?: string;
  dropdownMenuTitleActiveTextColor?: string;
  dropdownMenuTitleDisabledTextColor?: string;
  dropdownMenuTitlePadding?: string;
  dropdownMenuTitleLineHeight?: number | string;
  dropdownMenuOptionActiveColor?: string;
  dropdownMenuOptionDisabledColor?: string;
  dropdownMenuContentMaxHeight?: string;
};
