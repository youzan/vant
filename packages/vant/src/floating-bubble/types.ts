export type FloatingBubbleThemeVars = {
  floatingBubbleSize?: string;
  floatingBubbleInitialGap?: string;
  floatingBubbleIconSize?: string;
  floatingBubbleBackground?: string;
  floatingBubbleColor?: string;
  floatingBubbleZIndex?: number | string;
};

export type FloatingBubbleAxis = 'x' | 'y' | 'xy' | 'lock';

export type FloatingBubbleMagnetic = 'x' | 'y';

export type FloatingBubbleOffset = {
  x: number;
  y: number;
};

export type FloatingBubbleGap =
  | number
  | {
      x: number;
      y: number;
    };

export type FloatingBubbleBoundary = {
  top: number;
  right: number;
  bottom: number;
  left: number;
};
