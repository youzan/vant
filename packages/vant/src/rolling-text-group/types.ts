import { RollingTextGroupProps } from './RollingTextGroup';
import type { ComponentPublicInstance } from 'vue';

export type RollingTextGroupExpose = {
  start: () => void;
  reset: () => void;
};

export type RollingTextGroupInstance = ComponentPublicInstance<
  RollingTextGroupProps,
  RollingTextGroupExpose
>;
