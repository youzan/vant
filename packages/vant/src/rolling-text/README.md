# RollingText

### Introduction

Rolling text animation, which can roll numbers and other types of text. Please upgrade `vant` to >= v4.6.0 before using this component.

### Install

Register component globally via `app.use`, refer to [Component Registration](#/en-US/advanced-usage#zu-jian-zhu-ce) for more registration ways.

```js
import { createApp } from 'vue';
import { RollingText, RollingTextGroup } from 'vant';

const app = createApp();
app.use(RollingText);
app.use(RollingTextGroup);
```

## Usage

### Basic Usage

You can set the starting value with `start-num` and the target value with `target-num`. The RollingText component will automatically start the animation, rolling from the starting value to the target value.

```html
<van-rolling-text :start-num="0" :target-num="123" />
```

### Set Rolling Direction

You can set the rolling direction of the numbers using the `direction` prop. By default, it rolls downwards, but you can set it to `up` to roll upwards.

```html
<van-rolling-text :start-num="0" :target-num="432" direction="up" />
```

### Set Stop Order

You can set the order of stopping the animation of each digit through the `stop-order` prop. By default, it stops from the higher digits. Setting `rtl` can stop from the ones digit.

```html
<van-rolling-text :start-num="0" :target-num="54321" stop-order="rtl" />
```

### Roll Non-numeric Text

You can reverse non-numeric content by using the `text-list` prop. The component will rolling from the first item to the last item in the array. Please make sure that the array length is greater than or equal to 2, and that each item has the same length.

```html
<van-rolling-text :text-list="textList" :duration="1" />
```

```js
import { ref } from 'vue';

export default {
  setup() {
    const textList = ref([
      'aaaaa',
      'bbbbb',
      'ccccc',
      'ddddd',
      'eeeee',
      'fffff',
      'ggggg',
    ]);
    return { textList };
  },
};
```

### Custom Style

The RollingText component provides some CSS variables that you can override to customize the style, or you can directly modify the component's style. Additionally, you can set the height of the numbers using the `height` prop.

```html
<van-rolling-text
  class="my-rolling-text"
  :height="54"
  :start-num="12345"
  :target-num="54321"
/>
```

```css
.my-rolling-text {
  --van-rolling-text-background: #1989fa;
  --van-rolling-text-color: white;
  --van-rolling-text-font-size: 24px;
  --van-rolling-text-gap: 6px;
  --van-rolling-text-item-border-radius: 5px;
  --van-rolling-text-item-width: 40px;
}
```

### RollingText Group

You can use RollingTextGroup to control the animation of all RollingTexts, and add other element. Please upgrade vant to >= v4.9.0 before using this component.

```html
<rolling-text-group>
  <span style="display: inline;">-¥</span>
  <van-rolling-text :target-num="1" />
  <van-rolling-text :target-num="9" />
  <span style="display: inline;">.</span>
  <van-rolling-text :target-num="9" />
  <van-rolling-text :target-num="8" />
</rolling-text-group>
```

### Manual Control

After obtaining the component instance through `ref`, you can call the `start` and `reset` methods. The `start` method is used to start the animation, and the `reset` method is used to reset the animation.

```html
<van-rolling-text
  ref="rollingTextRef"
  :start-num="0"
  :target-num="54321"
  :auto-start="false"
/>
<van-grid clickable :column-num="3">
  <van-grid-item icon="play-circle-o" :text="start" @click="start" />
  <van-grid-item icon="replay" :text="reset" @click="reset" />
</van-grid>
```

```js
import { ref } from 'vue';

export default {
  setup() {
    const rollingTextRef = ref(null);
    const start = () => {
      rollingTextRef.value.start();
    };
    const reset = () => {
      rollingTextRef.value.reset();
    };
    return { rollingTextRef, start, reset };
  },
};
```

## API

### RollingText Props

| Attribute | Description | Type | Default |
| --- | --- | --- | --- |
| start-num | Start number | _number_ | `0` |
| target-num | Target number | _number_ | - |
| text-list | Text array | _string[]_ | `[]` |
| duration | Duration of the animation, in seconds | _number_ | `2` |
| direction | Rolling direction of the text, with `down` and `up` as the values | _string_ | `down` |
| auto-start | Whether to start the animation | _boolean_ | `true` |
| stop-order | Order of stopping the animation of each digit, with `ltr` and `rtl` as the values | _string_ | `ltr` |
| height | Height of digit, `px` as unit | _number_ | `40` |

### RollingTextGroup Props

| 参数 | 说明 | 类型 | 默认值 |
| --- | --- | --- | --- |
| start-num `v4.9.0` | Start number of all RollingTexts | _number_ | `0` |
| duration `v4.9.0` | Duration of all RollingTexts, in seconds | _number_ | `2` |
| direction `v4.9.0` | Rolling direction of all RollingTexts, with `down` and `up` as the values | _string_ | `down` |
| auto-start `v4.9.0` | Whether to start all RollingTexts | _boolean_ | `true` |
| stop-order `v4.9.0` | Order to stop all RollingTexts, with `ltr` and `rtl` as the values | _string_ | `ltr` |
| height `v4.9.0` | Height of all RollingTexts, `px` as unit | _number_ | `40` |

### RollingText Methods

Use [ref](https://vuejs.org/guide/essentials/template-refs.html) to get RollingText instance and call instance methods.

| Name  | Description         | Attribute | Return value |
| ----- | ------------------- | --------- | ------------ |
| start | Start the animation | -         | -            |
| reset | Reset the animation | -         | -            |

### RollingTextGroup Methods

Use [ref](https://vuejs.org/guide/essentials/template-refs.html) to get RollingTextGroup instance and call instance methods.

| Name           | Description         | Attribute | Return value |
| -------------- | ------------------- | --------- | ------------ |
| start `v4.9.0` | Start the animation | -         | -            |
| reset `v4.9.0` | Reset the animation | -         | -            |

### Types

The component exports the following type definitions:

```ts
import type {
  RollingTextProps,
  RollingTextInstance,
  RollingTextDirection,
  RollingTextStopOrder,
  RollingTextGroupProps,
  RollingTextGroupInstance,
} from 'vant';
```

`RollingTextInstance` and `RollingTextGroupInstance` is the type of component instance:

```ts
import { ref } from 'vue';
import type { RollingTextInstance, RollingTextGroupInstance } from 'vant';

const rollingTextRef = ref<RollingTextInstance>();
const rollingTextGroupRef = ref<RollingTextGroupInstance>();

rollingTextRef.value?.start();
rollingTextGroupRef.value?.start();
```

## Theming

### CSS Variables

The component provides the following CSS variables, which can be used to customize styles. Please refer to [ConfigProvider component](#/en-US/config-provider).

| Name | Default Value | Description |
| --- | --- | --- |
| --van-rolling-text-background | _inherit_ | Background color of a single digit |
| --van-rolling-text-color | _var(--van-text-color)_ | Color of the number |
| --van-rolling-text-font-size | _var(--van-font-size-md)_ | Font size of the number |
| --van-rolling-text-gap | _0px_ | Spacing between digits |
| --van-rolling-text-item-width | _15px_ | Width of a single digit |
| --van-rolling-text-item-border-radius | _0px_ | Border radius of a single digit |
