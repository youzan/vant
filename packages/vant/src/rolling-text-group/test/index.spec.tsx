import { nextTick } from 'vue';
import { mount } from '../../../test';
import { RollingTextGroup } from '../index';
import { RollingText } from '../../rolling-text';
import { RollingTextGroupInstance } from '../types';

const itemWrapperClass = '.van-rolling-text-item__box';
const animationClass = 'van-rolling-text-item__box--animate';

test('should render default slot correctly', async () => {
  const wrapper = mount({
    render() {
      return (
        <RollingTextGroup>
          <RollingText targetNum={1} />
          <span>.</span>
          <RollingText targetNum={2} />
        </RollingTextGroup>
      );
    },
  });
  await nextTick();
  expect(wrapper.html()).toMatchSnapshot();
});

test('should set start-num for all rollingTexts unless already specified', async () => {
  const wrapper = mount({
    render() {
      return (
        <RollingTextGroup startNum={1}>
          <RollingText targetNum={1} />
          <RollingText startNum={2} targetNum={2} />
        </RollingTextGroup>
      );
    },
  });
  await nextTick();
  const boxes = wrapper.findAll('.van-rolling-text-item__box');

  const firstRollingText = boxes[0]
    .findAll('.van-rolling-text-item__item')[0]
    .text();
  expect(firstRollingText).toEqual('1');

  const secondRollingText = boxes[1]
    .findAll('.van-rolling-text-item__item')[0]
    .text();
  expect(secondRollingText).toEqual('2');
});

test('should set duration for all rollingTexts unless already specified', async () => {
  const wrapper = mount({
    render() {
      return (
        <RollingTextGroup duration={1}>
          <RollingText targetNum={1} />
          <RollingText duration={0.5} targetNum={2} />
        </RollingTextGroup>
      );
    },
  });
  await nextTick();
  const rollingTexts = wrapper.findAll('.van-rolling-text-item');

  expect(rollingTexts[0].style.getPropertyValue('--van-duration')).toEqual(
    '1s',
  );
  expect(rollingTexts[1].style.getPropertyValue('--van-duration')).toEqual(
    '0.5s',
  );
});

test('should set direction for all rollingTexts unless already specified', async () => {
  const wrapper = mount({
    render() {
      return (
        <RollingTextGroup direction={'up'}>
          <RollingText targetNum={1} />
          <RollingText direction={'down'} targetNum={2} />
        </RollingTextGroup>
      );
    },
  });
  await nextTick();
  const rollingTexts = wrapper.findAll('.van-rolling-text-item');

  expect(rollingTexts[0].classes()).toContain('van-rolling-text-item--up');
  expect(rollingTexts[1].classes()).toContain('van-rolling-text-item--down');
});

test('should set stopOrder for all rollingTexts unless already specified', async () => {
  const wrapper = mount({
    render() {
      return (
        <RollingTextGroup stopOrder={'rtl'}>
          <RollingText stopOrder={'ltr'} targetNum={1} />
          <RollingText targetNum={2} />
          <RollingText targetNum={3} />
        </RollingTextGroup>
      );
    },
  });
  await nextTick();
  const rollingTexts = wrapper.findAll('.van-rolling-text-item');

  expect(rollingTexts[0].style.getPropertyValue('--van-delay')).toEqual('0s');
  expect(rollingTexts[1].style.getPropertyValue('--van-delay')).toEqual('0.2s');
  expect(rollingTexts[2].style.getPropertyValue('--van-delay')).toEqual('0s');
});

test('should set height for all rollingTexts unless already specified', async () => {
  const wrapper = mount({
    render() {
      return (
        <RollingTextGroup height={50}>
          <RollingText targetNum={1} />
          <RollingText height={100} targetNum={2} />
        </RollingTextGroup>
      );
    },
  });
  await nextTick();
  const boxes = wrapper.findAll('.van-rolling-text-item__box');

  const firstRollingText = boxes[0].findAll('.van-rolling-text-item__item')[0];
  expect(firstRollingText.style.lineHeight).toEqual('50px');

  const secondRollingText = boxes[1].findAll('.van-rolling-text-item__item')[0];
  expect(secondRollingText.style.lineHeight).toEqual('100px');
});

test('should control the animation of all rollingTexts', async () => {
  const wrapper = mount<RollingTextGroupInstance>(RollingTextGroup, {
    props: {
      autoStart: false,
    },
    slots: {
      default: () => (
        <>
          <RollingText targetNum={1} />
          <RollingText targetNum={2} />
        </>
      ),
    },
  });
  const instance = wrapper.vm;
  instance.start();
  await nextTick();
  expect(wrapper.find(itemWrapperClass).classes()).toContain(animationClass);

  instance.reset();
  await nextTick();
  expect(wrapper.find(itemWrapperClass).classes()).not.contain(animationClass);
});
