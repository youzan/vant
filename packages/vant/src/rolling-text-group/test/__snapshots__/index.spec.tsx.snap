// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render default slot correctly 1`] = `
<div class="van-rolling-text-group">
  <div class="van-rolling-text">
    <div
      class="van-rolling-text-item van-rolling-text-item--down"
      style="height: 40px; --van-translate: -1640px; --van-duration: 2s; --van-delay: 0s;"
    >
      <div class="van-rolling-text-item__box van-rolling-text-item__box--animate">
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          1
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          0
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          9
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          8
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          7
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          6
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          5
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          4
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          3
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          2
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          1
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          0
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          9
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          8
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          7
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          6
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          5
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          4
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          3
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          2
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          1
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          0
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          9
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          8
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          7
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          6
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          5
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          4
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          3
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          2
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          1
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          0
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          9
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          8
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          7
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          6
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          5
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          4
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          3
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          2
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          1
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          0
        </div>
      </div>
    </div>
  </div>
  <span>
    .
  </span>
  <div class="van-rolling-text">
    <div
      class="van-rolling-text-item van-rolling-text-item--down"
      style="height: 40px; --van-translate: -1680px; --van-duration: 2s; --van-delay: 0.2s;"
    >
      <div class="van-rolling-text-item__box van-rolling-text-item__box--animate">
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          2
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          1
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          0
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          9
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          8
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          7
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          6
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          5
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          4
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          3
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          2
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          1
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          0
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          9
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          8
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          7
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          6
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          5
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          4
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          3
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          2
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          1
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          0
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          9
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          8
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          7
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          6
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          5
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          4
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          3
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          2
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          1
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          0
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          9
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          8
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          7
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          6
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          5
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          4
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          3
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          2
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          1
        </div>
        <div
          class="van-rolling-text-item__item"
          style="line-height: 40px;"
        >
          0
        </div>
      </div>
    </div>
  </div>
</div>
`;
